File: AD-HOC/2444/main.js
let lines = require('fs').readFileSync('/dev/stdin', 'utf-8').trim().split('\n');
let [v, t] = lines.shift().split(' ').map(Number);
let a = lines.shift().split(' ').map(Number);
for (let i = 0; i < t; i++) {
    v += a[i];
    if (v < 0) v = 0;
    else if (v > 100) v = 100;
}
console.log(v);

File: AD-HOC/2444/main.dart
import 'dart:io';

void main() {
  var [v, t] = stdin.readLineSync()!.split(' ').map(int.parse).toList();
  List<int> a = stdin.readLineSync()!.split(' ').map(int.parse).toList();
  for (int i = 0; i < t; i++) {
    v += a[i];
    if (v < 0)
      v = 0;
    else if (v > 100) v = 100;
  }
  print(v);
}
